src/configuration.c

```diff
@@ -132,8 +132,22 @@ bool read_config_from_file(char *config_file, struct configuration *config)
                                         config->boolValues[i] = true;
                                 }
                         } else {
-                                fprintf(stderr, "ERR: missing key with boolean value: %s\n", config_key_bool[i]);
-                                return false;
+                                /* New configuration keys, CONF_RECORD_RETENTION_ENABLED and CONF_RECORD_SERVER_DELIVERY_ENABLED
+                                 * default values, otherwise update will require changes in custom configurations */
+                                if (i == CONF_RECORD_RETENTION_ENABLED) {
+#ifdef DEBUG
+                                        fprintf(stderr, "WARN: missing boolean optional key: %s\n", config_key_bool[i]);
+#endif
+                                        config->boolValues[i] = RECORD_RETENTION_ENABLED_DEFAULT;
+                                } else if (i == CONF_RECORD_SERVER_DELIVERY_ENABLED) {
+#ifdef DEBUG
+                                        fprintf(stderr, "WARN: missing boolean optional key: %s\n", config_key_bool[i]);
+#endif
+                                        config->boolValues[i] = RECORD_SERVER_DELIVERY_ENABLED_DEFAULT;
+                                } else {
```

Comment:
Not sure this `else` case makes sense... Wouldn't this just catch an error in the code?

Reply:
The type of errors that are caught in this block are configuration errors, the specific error is a required boolean key missing. Before this change all the keys in a configuration were "required" and if one was missing the daemon printed an error and quit.

This change adds an ```if, else if, and else``` statement.  The first two are to cover for the "optional keys" ```record_server_delivery_enabled``` and ```record_retention_enabled``` when there is no explicit key defined in configuration a default value is assigned (to maintain backward compatibility with existing custom configurations). For all the other keys the previous behavior is maintained "print error and quit".

Reply:
Thanks, that makes sense now. This block of code is executed when the non-optional keys are missing, and that's still expected behavior.

Disregard my comment :-)